homeassistant/components/flux_led/light.py

```diff
@@ -171,133 +141,231 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
+
+    def __init__(
+        self,
+        coordinator: FluxLEDCoordinator,
+        unique_id: str,
+        device: dict,
+        effect_speed: int,
+        hass: HomeAssistant,
+    ):
+        """Initialize the Flux light entity."""
+        super().__init__(coordinator=coordinator)
+
+        self._name = device[CONF_NAME]
+        self._unique_id = unique_id
+        self._icon = "mdi:lightbulb"
+        self._attrs = {}
+        self._last_update = 0
+        self._state = None
+        self._brightness = None
+        self._hs_color = None
+        self._bulb = coordinator.light
+        self._last_brightness = 255
+        self._last_hs_color = color_util.color_RGB_to_hs(255, 255, 255)
+        self._ip_address = device[CONF_HOST]
+        self._effect_speed = effect_speed
+        self._mode = coordinator.data["mode"]
+
+        async def async_remove_light(device: dict):
+            """Remove a bulb device when it is removed from options."""
+
+            bulb_id = device["device_id"]
+
+            if self._unique_id != bulb_id:
+                return
+
+            entity_registry = await hass.helpers.entity_registry.async_get_registry()
+            entity_entry = entity_registry.async_get(self.entity_id)
+
+            device_registry = await hass.helpers.device_registry.async_get_registry()
+            device_entry = device_registry.async_get(entity_entry.device_id)
+
+            if (
+                len(
+                    async_entries_for_device(
+                        entity_registry,
+                        entity_entry.device_id,
+                        include_disabled_entities=True,
+                    )
+                )
+                == 1
+            ):
+                device_registry.async_remove_device(device_entry.id)
 
-        # After bulb object is created the status is updated. We can
-        # now set the correct mode if it was not explicitly defined.
-        if not self._mode:
-            if self._bulb.rgbwcapable:
-                self._mode = MODE_RGBW
-            else:
-                self._mode = MODE_RGB
+            entity_registry.async_remove(self.entity_id)
 
-    def _disconnect(self):
-        """Disconnect from Flux light."""
-        self._bulb = None
+        async_dispatcher_connect(hass, SIGNAL_REMOVE_DEVICE, async_remove_light)
 
     @property
-    def available(self) -> bool:
-        """Return True if entity is available."""
-        return self._bulb is not None
+    def unique_id(self):
+        """Return the unique ID of the light."""
+        return self._unique_id
 
     @property
     def name(self):
-        """Return the name of the device if any."""
+        """Return the name of the light."""
         return self._name
 
     @property
     def is_on(self):
-        """Return true if device is on."""
-        return self._bulb.isOn()
+        """Return true if the light is on."""
+
+        if time.time() - self._last_update > 1:
```

Comment:
Why do we need this check? The coordinator data should be the source of truth.

Reply:
There is a slight delay between the state change and the change reporting correctly in the coordinator (from the light). Without this the UI switch flips back and forth when it is triggered on or off.

Reply:
If we don't get the feedback fast enough we should update to the assumed state in the actuator method.

Reply:
I'm assuming you mean when the light is triggered on/off from Home Assistant side.

Reply:
Yes - when the light is triggered from Home Assistant.

I've been trying to figure out how to update the assumed state - is there a good example you can point me to?

Reply:
Update the the state holding instance attribute, eg `self._state` and schedule an entity state update with eg `self.async_write_ha_state`.

Reply:
OK - got it. Updated and removed the state delays.